auth/src/main/java/org/aerogear/auth/credentials/OIDCCredentials.java

```diff
@@ -67,8 +145,21 @@ public boolean isAuthorized() {
      * Returns stringified JSON for the OIDCCredential.
      * @return Stringified JSON OIDCCredential
      */
-    public String serialise() {
-        return this.authState.jsonSerializeString();
+    public String serialize() throws JSONException {
+        JSONObject jsonCredential = new JSONObject()
+            .put("authState", this.authState.jsonSerializeString());
+        if (this.integrityCheckParameters != null) {
+            jsonCredential.put("integrityCheck", this.integrityCheckParameters.serialize());
+        }
+        return jsonCredential.toString();
+    }
+
+    public static OIDCCredentials deserialize(final String serializedCredential) throws JSONException {
```

Comment:
If this object is supposed to be created only through this method, the constructor should be private.